From 8dd3c317991171ff8c0d7e341c6194577e73a6a3 Mon Sep 17 00:00:00 2001
From: StanleyYP Wang <StanleyYP.Wang@mediatek.com>
Date: Fri, 14 Feb 2025 15:47:13 +0800
Subject: [PATCH 092/104] mtk: mt76: mt7996: add the missing testmode
 compilation flag

Add the missing testmode compilation flag to prevent build error when
CONFIG_NL80211_TESTMODE is not defined.

Signed-off-by: StanleyYP Wang <StanleyYP.Wang@mediatek.com>
---
 mt76.h               |  9 +++++++++
 mt7996/main.c        |  8 ++++++--
 mt7996/mcu.c         | 11 +++++------
 mt7996/mtk_debugfs.c |  2 +-
 4 files changed, 21 insertions(+), 9 deletions(-)

diff --git a/mt76.h b/mt76.h
index bbe1af77..04e7198e 100644
--- a/mt76.h
+++ b/mt76.h
@@ -1680,6 +1680,15 @@ static inline bool mt76_testmode_enabled(struct mt76_phy *phy)
 #endif
 }
 
+static inline bool mt76_testmode_bf_enabled(struct mt76_phy *phy)
+{
+#ifdef CONFIG_NL80211_TESTMODE
+	return phy->test.bf_en;
+#else
+	return false;
+#endif
+}
+
 static inline bool mt76_is_testmode_skb(struct mt76_dev *dev,
 					struct sk_buff *skb,
 					struct ieee80211_hw **hw)
diff --git a/mt7996/main.c b/mt7996/main.c
index 536695fd..c74b55b8 100644
--- a/mt7996/main.c
+++ b/mt7996/main.c
@@ -12,6 +12,7 @@
 
 static void mt7996_testmode_disable_all(struct mt7996_dev *dev)
 {
+#ifdef CONFIG_NL80211_TESTMODE
 	struct mt7996_phy *phy;
 	int i;
 
@@ -20,6 +21,7 @@ static void mt7996_testmode_disable_all(struct mt7996_dev *dev)
 		if (phy)
 			mt76_testmode_set_state(phy->mt76, MT76_TM_STATE_OFF);
 	}
+#endif
 }
 
 int mt7996_run(struct mt7996_phy *phy)
@@ -582,8 +584,10 @@ static void mt7996_remove_interface(struct ieee80211_hw *hw,
 		if (vif == phy->monitor_vif) {
 			phy->monitor_vif = NULL;
 
+#ifdef CONFIG_NL80211_TESTMODE
 			if (dev->testmode_enable)
 				kfree(phy->mt76->lists);
+#endif
 		}
 	}
 
@@ -609,7 +613,7 @@ int mt7996_set_channel(struct mt76_phy *mphy)
 	int ret = 0;
 
 	if (mphy->chanctx && mphy->chanctx->state == MT76_CHANCTX_STATE_ADD) {
-		if (!mt76_testmode_enabled(phy->mt76) && !phy->mt76->test.bf_en) {
+		if (!mt76_testmode_enabled(mphy) && !mt76_testmode_bf_enabled(mphy)) {
 			ret = mt7996_mcu_edcca_enable(phy, true);
 			if (ret)
 				goto out;
@@ -637,7 +641,7 @@ int mt7996_set_channel(struct mt76_phy *mphy)
 			goto out;
 	}
 
-	if (mt76_testmode_enabled(phy->mt76) || phy->mt76->test.bf_en) {
+	if (mt76_testmode_enabled(mphy) || mt76_testmode_bf_enabled(mphy)) {
 		mt7996_tm_update_channel(phy);
 		goto out;
 	}
diff --git a/mt7996/mcu.c b/mt7996/mcu.c
index c600109a..fbf96cb3 100644
--- a/mt7996/mcu.c
+++ b/mt7996/mcu.c
@@ -1945,12 +1945,11 @@ mt7996_mcu_bss_basic_tlv(struct sk_buff *skb, struct ieee80211_vif *vif,
 	bss->hw_bss_idx = idx;
 
 	if (vif->type == NL80211_IFTYPE_MONITOR) {
-		struct mt76_testmode_data *td = &phy->test;
-
-		if (!td->bf_en)
-			memcpy(bss->bssid, phy->macaddr, ETH_ALEN);
-		else
-			memcpy(bss->bssid, td->addr[2], ETH_ALEN);
+		memcpy(bss->bssid, phy->macaddr, ETH_ALEN);
+#ifdef CONFIG_NL80211_TESTMODE
+		if (mt76_testmode_bf_enabled(phy))
+			memcpy(bss->bssid, phy->test.addr[2], ETH_ALEN);
+#endif
 		return 0;
 	}
 
diff --git a/mt7996/mtk_debugfs.c b/mt7996/mtk_debugfs.c
index 89f3506e..cbc299f0 100644
--- a/mt7996/mtk_debugfs.c
+++ b/mt7996/mtk_debugfs.c
@@ -2906,8 +2906,8 @@ static const struct file_operations mt7996_txpower_path_fops = {
 static int mt7996_show_eeprom_mode(struct seq_file *s, void *data)
 {
 	struct mt7996_dev *dev = dev_get_drvdata(s->private);
-	struct mt76_dev *mdev = &dev->mt76;
 #ifdef CONFIG_NL80211_TESTMODE
+	struct mt76_dev *mdev = &dev->mt76;
 	const char *mtd_name = mdev->test_mtd.name;
 	u32 mtd_offset = mdev->test_mtd.offset;
 #else
-- 
2.45.2

